### DISCUSSION TITLE
Can I toggle cacheAsBitmap on run-time on the fly?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7752] [Q&A]
It seems I can.
But I'm bit worried about the moment I set cacheAsBitmap to true about performance kind of cpu spikes.
Currently it looks everything is ok.
What happens if I toggle a lot of sprites or large container at same time?

**Comment by miltoncandelero** [comment]
As you can see, setting `cacheAsBitmap` to something different that what it was will trigger a destruction or creation of the bitmap cache: https://github.com/pixijs/pixijs/blob/dev/packages/mixin-cache-as-bitmap/src/index.ts#L145

**Comment by themoonrat** [comment]
Toggling cacheAsBitmap could be okay, or might not be okay, based on what it is you're trying to cache and how often you are doing it. What might be acceptable for users of one app might not be acceptable for users of another app, so it's impossible to answer if it's okay to do or not! Turning it on does have a one off performance cost, plus ongoing memory cost, for the benefit of reduced ongoing gpu cost. It's a balancing act.

What I would say is, if you're finding you are needing to cache a certain part frequently, that maybe you can create your own render textures yourself at a convenient time (app start, or during loading screen, or change of level etc.)  and use those. Render textures are what cacheAsBitmap uses in the background, after all.
